### DISCUSSION TITLE
Interest in Microsoft Azure as a new Server Provider

### DISCUSSION THREAD

**Discussion by cloudchristoph** [vitodeploy/vito#361] [Polls]
Is there any interest in the community in extending the server providers with Microsoft Azure? 
I am currently implementing this rudimentarily in my fork of the 2.x version and wonder if others are also interested or if Azure is not needed.

Just an idea, of course. If the maintainers say it won't happen, even if someone else implements it, that's perfectly fine.

Best,
Christoph

**Comment by ahaneef29** [comment]
I have developed and deployed laravel app in azure and if this service provider is added with Vito! it would be great chance for laravel/php developer to explore azure web app in near future.

Looking forward

**Comment by saeedvaziry** [comment]
We'll certainly add Azure as a server provider, it's just a matter of priority now.

**Reply by cloudchristoph** [reply]
I understand that completely. That's why I'm going to try it in my fork first and see how far I get.
If everything works out, then I will continue to look after this provider. I'm a Cloud Architect for Azure on a day-to-day basis anyway - so it's an obvious choice. 😉

**Comment by joglomedia** [comment]
Hi all,

I am currently working on a PoC for the Azure provider, but I am consistently encountering API errors:

- `Azure API: NIC(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another NIC(s) or retry after 180 seconds.`
- `A retryable error occurred.`

The Azure documentation is not very clear on this matter. Based on my current understanding of these issues:

For the NIC-related error, it appears that the creation of the server should be delayed until the network interface is fully prepared. The error message suggests a delay of 180 seconds (3 minutes) after the creation of the network interface.

Regarding the retryable error, it may be related to API throttling.

### Steps to Reproduce
1) Create a `publicIPAddress` from `providers/Microsoft.Network/publicIPAddresses`.
2) Create a `networkSecurityGroups` from `providers/Microsoft.Network/networkSecurityGroups`.
3) Create a `virtualNetworks` from `providers/Microsoft.Network/virtualNetworks`.
4) Create a `networkInterfaces` from `providers/Microsoft.Network/networkInterfaces`.
5) Create a `virtualMachines` from `providers/Microsoft.Compute/virtualMachines`.

### Additional Context
Should we use queued job for virtual machine creation? Any pros & cons of this method?
Or am I missing something for creating virtual machine?

### Screenshot
![image](https://github.com/user-attachments/assets/b2c236bd-5c52-44e5-9c8f-2ac440165301)

**Comment by saeedvaziry** [comment]
Do you have a PR @cloudchristoph @joglomedia for Azure? if yes I would like to test it 🚀

**Comment by saeedvaziry** [comment]
I suggest developing a plugin for this

https://vitodeploy.com/docs/plugins
